<launch>
<!-- %%%%%%%%%% ATOM Calibration Framework %%%%%%%%%%%%%%-->
<!-- {{ c.filename }} : Plays back the bag file. Sets up image decompressors if needed, reads the urdf robot description -->
<!-- This file was generated automatically by the script configure_calibration_pkg on the {{ c.date }}-->
<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->

<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<!-- Parameters-->
<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<!-- bag_file: you can give another bagfile to the launch file using this arg-->
<arg name="bag_file" default="{{ c.bag_file }}"/>
<!-- bag_start: starting time for playing back the bag file. Check "rosbag play -h"-->
<arg name="bag_start" default="0"/>
<arg name ="bag_rate" default ="1" />
<arg name="description_file" default="$(find {{ c.package_name }})/urdf/description.urdf.xacro"/>
<arg name="rviz_file" default="$(find {{ c.package_name }})/rviz/{{ c.rviz_set_initial_estimate }}"/>

<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<!-- Load robot description and tf generators -->
<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<param name = "robot_description" command = "$(find xacro)/xacro $(arg description_file)" />
<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<!-- Playback the bag file -->
<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<param name="/use_sim_time" value="true"/>
<node pkg="rosbag" type="play" name="rosbag_play" output="screen" args=" $(arg bag_file) --clock -r $(arg bag_rate) -l -s $(arg bag_start) /tf:=/tf_dev_null /tf_static:=/tf_static_dev_null"/>

{% if c.use_compressed_topics %}
<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<!-- Image Topic Decompression -->
<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
{% for key, value in c.compressed_topics.iteritems() -%}
<node pkg="image_transport" type="republish" name="republish_{{ value['sensor_key'] }}" output="screen" args="compressed in:={{ key }} raw out:={{ key }}"/>
{% endfor %}
{% endif %}

<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<!-- Visualization -->
<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%-->
<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_file)" required="true"/>

</launch>